terraform-zerotrust-postgrest-worker/README.md

```diff
@@ -0,0 +1,90 @@
+    # optional, defaults to following values
+    export TF_VAR_gcp_zone=us-east1-b
+    export TF_VAR_gcp_machine_type=f1-micro
+    export TF_VAR_cloudflare_ssh_subdomain=zerotrust-ssh-example
+    export TF_VAR_cloudflare_postgrest_subdomain=zerotrust-postgrest-example
+    export TF_VAR_cloudflare_worker_subdomain=zerotrust-worker-example
+    ```
+4. Run `terraform apply` to initiate config
+
+## Requirements
+
+- Terraform version 0.13 +
+- GCP (Google Cloud) cli setup and [authenticated to a GCP project](https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login) with compute access. 
+- Cloudflare account and zone with at least 1 Access seat
+
+## Setup
+
+Typically takes ~5 mins to fully spin up. 
+
+This demo 
+- creates a Google Cloud (GCP) instance running an [PostgreSQL](https://hub.docker.com/_/postgres) and [PostgREST](https://hub.docker.com/r/postgrest/postgrest/) docker container.
+- spins up a Cloudflare Tunnel on the GCP instance proxying traffic to the PostgREST container
+and the local SSH port on the server
+- deploys Cloudflare Worker (+route) that is using Cloudflare Access service tokens to access the protected PostgREST endpoint
+- creates DNS records at Cloudflare to point to the tunnel endpoints and for the [example Worker](./cloudflare-worker-script.js)
+
+The GCP instance is set to not accept SSH traffic via a `no-ssh` tag. Your Firewall settings may differ and this may need to be adjusted as needed.
+
+An access policy is created for 
+- `zerotrust-ssh-example.yourzone.com` _(or your configured subdomain via `TF_VAR_cloudflare_ssh_subdomain` variable)_ 
+- `zerotrust-postgrest-example.yourzone.com` _(or your configured subdomain via `TF_VAR_cloudflare_postgrest_subdomain` variable)_
+- `zerotrust-worker-example.yourzone.com` _(or your configured subdomain via `TF_VAR_cloudflare_worker_subdomain` variable)_
+
+The only person allowed to all of them is the user provided to `TF_VAR_cloudflare_email` environment variable.
+
+## Zero Trust access to your instance
+### local SSH
+Update your workstation's SSH config to provide a specific host block for `zerotrust-ssh-example.yourzone.com`.
+
+Here is an example:
+
+```
+Host zerotrust-ssh-example.yourzone.com
+    IdentityFile /Users/user/.ssh/google_compute_engine
+    ProxyCommand /usr/local/bin/cloudflared access ssh --hostname %h
+```
+
+You can then `ssh user@zerotrust-ssh-example.yourzone.com` where user is your OS user to login.
+
+## Endpoints
+
+Where `yourzone.com` is the zone assigned to `TF_VAR_cloudflare_zone` environment variable.
+
+- `zerotrust-ssh-example.yourzone.com` Access protected for SSH access, exposed with Cloudflare Tunnel
+- `zerotrust-postgrest-example.yourzone.com` Access PostgREST (PostgreSQL proxy), exposed with Cloudflare Tunnel
+- `zerotrust-worker-example.yourzone.com` publicly available [example Worker](./cloudflare-worker-script.js) querying PostgREST from the Cloudflare Edge, using Cloudflare Access service token
```

Comment:
what's the advantage of having this Worker on top of the tunnel, versus calling the tunnel's hostname directly myself?

Reply:
The PostgREST is not likely to be accessed by users directly, and I don't wanna them to have access to the PostgREST either. In my case, I have our application running on Workers completely, so I can ingest the data after some kind of user authentication.

Reply:
Also, its very easy to leverage the Workers ecosystem, for example, KV/Cache API to cache the database reads. :)

Reply:
I see. So PostgREST allows arbitrary access to the database, so it is protected by Access. The Worker exposes subset of the database functionality, so it is open to the world.

I understand you had already written the README to try to convey this idea, but the "responsibility split" between PostgREST and the Worker were not clear to me, so maybe you can slightly improve the text to convey that?

Reply:
@aaranmcguire done, tried to improve the text a bit :) feel free to suggest any changes